src/xero/basemanager.py

```diff
@@ -440,11 +479,30 @@ def generate_param(key, value):
                     field = field.replace("_", ".")
                 return fmt % (field, get_filter_params(key, value))
 
+            KNOWN_PARAMETERS = ["order", "offset", "page"]
+            object_params = self.OBJECT_FILTER_FIELDS.get(self.name, {})
+            LIST_PARAMETERS = list(
+                filter(lambda x: object_params[x] == list, object_params)
+            )
+            EXTRA_PARAMETERS = list(
+                filter(lambda x: object_params[x] != list, object_params)
+            )
+
             # Move any known parameter names to the query string
-            KNOWN_PARAMETERS = ["order", "offset", "page", "includeArchived"]
-            for param in KNOWN_PARAMETERS:
+            for param in KNOWN_PARAMETERS + EXTRA_PARAMETERS:
+                if param in kwargs:
+                    params[param] = get_filter_value(
+                        param, kwargs.pop(param), object_params.get(param, None)
+                    )
+            # Support xero optimised list filtering; validate IDs we send but may need other validation
+            for param in LIST_PARAMETERS:
                 if param in kwargs:
-                    params[param] = kwargs.pop(param)
+                    if param.endswith("IDs"):
+                        params[param] = ",".join(
+                            map(lambda x: UUID(x).hex, kwargs.pop(param))
+                        )
+                    else:
+                        params[param] = ",".join(kwargs.pop(param))
```

Comment:
Processing of list filter fields, with no input verification except for IDs that are interpreted as UUIDs.